new/prusti-specs/src/specifications/preparser.rs

```diff
@@ -0,0 +1,584 @@
+use proc_macro2::{Delimiter, Group, Spacing, Span, TokenStream, TokenTree, Ident, };
+use std::collections::VecDeque;
+use std::mem;
+use syn::parse::{ParseBuffer, ParseStream, Parse};
+use syn::{self, Token, PatType, Pat, Error};
+use quote::ToTokens;
+
+use super::common;
+use crate::specifications::common::AssertionKind::ForAll;
+use crate::specifications::common::{ForAllVars, TriggerSet, Trigger};
+use syn::spanned::Spanned;
+
+pub type AssertionWithoutId = common::Assertion<(), syn::Expr, common::Arg>;
+pub type ExpressionWithoutId = common::Expression<(), syn::Expr>;
+
+#[derive(Debug)]
+struct ParserStream {
+    last_span: Span,
+    tokens: VecDeque<TokenTree>,
+}
+
+impl ParserStream {
+    fn empty() -> Self {
+        Self {
+            tokens: VecDeque::new(),
+            last_span: Span::call_site(),
+        }
+    }
+    fn from_token_stream(tokens: TokenStream) -> Self {
+        let token_queue: VecDeque<_> = tokens.into_iter().collect();
+        Self {
+            tokens: token_queue,
+            last_span: Span::call_site(),
+        }
+    }
+
+    /// Check if there is a subexpression that contains both and and or
```

Comment:
```suggestion
    /// Check if there is a subexpression that contains both `&&` and `||`.
```
Could you also explain in the comment why is this needed?